controllers/ciresource_fsm.go

```diff
@@ -217,17 +217,26 @@ func (f *CIResourceFSM) handleStateCleaningWait(context CIResourceFSMContext) (t
 
 func (f *CIResourceFSM) handleStateDelete(context CIResourceFSMContext) (time.Duration, error) {
 
-	f.logger.Info("removing resource", "Id", context.CIResource.Status.ResourceId)
+	if context.CIPool.IsFallbackPool() && context.CIResource.Status.ResourceId == "" {
```

Comment:
How could happen a scenario like that? Unless there's a bug a in the provider (or someone manually alters the resource field), I wouldn't expect to happen in a realistic case.